contracts/sdk/indexer/indexer/utils.ts

```diff
@@ -124,97 +62,272 @@ export function loadProgram(
+  const buffer = bs58.decode(base58String);
+  const layout = layouts.get(eventName);
+  if (!layout) {
+    console.error("Could not find corresponding layout for event:", eventName);
   }
-  const parsedLogs = parseLogs(logs.logs);
-  if (parsedLogs.length == 0) {
-    return;
+  const data = layout.decode(buffer);
+  return { data, name: eventName };
+}
+
+export function destructureBubblegumMintAccounts(
+  accountKeys: PublicKey[],
+  instruction: CompiledInstruction
+) {
+  return {
+    owner: accountKeys[instruction.accounts[4]],
+    delegate: accountKeys[instruction.accounts[5]],
+    merkleSlab: accountKeys[instruction.accounts[6]],
   }
-  db.connection.db.serialize(() => {
-    db.beginTransaction();
-    for (const parsedLog of parsedLogs) {
-      indexParsedLog(
-        db,
-        { txId: logs.signature, startSeq, endSeq },
-        context.slot,
-        parsedState,
-        parsedLog
-      );
+}
+
+
+type ZippedInstruction = {
+  instructionIndex: number,
+  instruction: CompiledInstruction,
+  innerInstructions: CompiledInnerInstruction[],
+}
+
+/// Similar to `order_instructions` in `/nft_ingester/src/utils/instructions.rs`
+function zipInstructions(
+  instructions: CompiledInstruction[],
+  innerInstructions: CompiledInnerInstruction[],
+): ZippedInstruction[] {
+  const zippedIxs: ZippedInstruction[] = [];
+  let innerIxIndex = 0;
+  for (let instructionIndex = 0; instructionIndex < instructions.length; instructionIndex++) {
+    const innerIxs = [];
+    while (innerIxIndex < innerInstructions.length && innerInstructions[innerIxIndex].index <= instructionIndex) {
+      innerIxs.push(innerInstructions[innerIxIndex]);
+      innerIxIndex += 1;
     }
-    db.commit();
-  });
+    zippedIxs.push({
+      instructionIndex,
+      instruction: instructions[instructionIndex],
+      innerInstructions: innerIxs
+    })
+  }
+  return zippedIxs;
 }
 
-/**
- * Processes the logs from a new transaction and searches for the programs
- * specified in the ParserState
- * @param db
- * @param logs
- * @param context
- * @param parsedState
- * @param startSeq
- * @param endSeq
- * @returns
- */
-export async function handleLogs(
+export function handleInstructionsAtomic(
   db: NFTDatabaseConnection,
-  logs: Logs,
+  instructionInfo: {
+    accountKeys: PublicKey[],
+    instructions: CompiledInstruction[],
+    innerInstructions: CompiledInnerInstruction[],
+  },
+  txId: string,
   context: Context,
   parsedState: ParserState,
   startSeq: number | null = null,
   endSeq: number | null = null
 ) {
-  if (logs.err) {
-    return;
-  }
-  const parsedLogs = parseLogs(logs.logs);
-  if (parsedLogs.length == 0) {
-    return;
-  }
-  for (const parsedLog of parsedLogs) {
-    await indexParsedLog(
+  const { accountKeys, instructions, innerInstructions } = instructionInfo;
+
+  const zippedInstructions = zipInstructions(instructions, innerInstructions);
+  for (let i = 0; i < zippedInstructions.length; i++) {
```

Comment:
`for (const zippedInstruction of zippedInstructions)` ?

Reply:
fixed